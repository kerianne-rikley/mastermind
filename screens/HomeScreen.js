import * as React from "react";
import { StyleSheet, View, Text, Button } from "react-native";

export default class HomeScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.titleCard}>Mastermind</Text>
        <View style={styles.buttonBox}>
          <Button 
            testID="start-button"
            title="Start Game"
            onPress={() => this.props.navigation.navigate("Game")}
          />
        </View>
        <View style={styles.spacer}/>
        <View style={styles.buttonBox}>
          <Button
            testID="how-to-button"
            title="How To Play"
            onPress={() => this.props.navigation.navigate("HowTo")}
          />
        </View>
      </View>
      );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center"
  },
  titleCard: {
    fontSize: 30,
    fontWeight: "bold"
  },
  buttonBox: {
    width: 120
  },
  spacer: {
    height: 2
  },
});
