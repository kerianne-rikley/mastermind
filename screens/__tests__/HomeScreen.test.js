import React from 'react';
import TestRenderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

import HomeScreen from '../HomeScreen';

describe('<HomeScreen />', () => {

    const mockNavigation = {
        navigate: jest.fn()
    }

    it('renders correctly', () => {
        const tree = TestRenderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('start button', () => {
        let startButton;

        beforeAll(() => {
            const { getByTestId } = render(<HomeScreen navigation={mockNavigation} />);
            startButton = getByTestId('start-button');
        });

        it('exists', () => {
            expect(startButton).toBeDefined();
        });

        it('navigates to game when clicked', () => {
            fireEvent.press(startButton);
            expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
            expect(mockNavigation.navigate).toHaveBeenCalledWith('Game');
        });
    });
});
