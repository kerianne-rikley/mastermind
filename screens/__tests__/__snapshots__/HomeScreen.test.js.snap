// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Mastermind
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="start-button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Start Game
      </Text>
    </View>
  </View>
</View>
`;
